src/dsl/verifier/proxy/parseBody.ts

```diff
@@ -4,24 +4,18 @@ interface ReqBodyExtended extends http.IncomingMessage {
   body?: Buffer | Record<string, unknown>;
 }
 
-export const parseBody = (
-  proxyReq: http.ClientRequest,
-  req: ReqBodyExtended
-): void => {
+export const parseBody = (req: ReqBodyExtended): Buffer => {
+  let bodyData = Buffer.alloc(0);
+
   if (!req.body || !Object.keys(req.body).length) {
-    return;
+    return bodyData;
   }
 
-  let bodyData;
-
   if (Buffer.isBuffer(req.body)) {
     bodyData = req.body;
   } else if (typeof req.body === 'object') {
-    bodyData = JSON.stringify(req.body);
+    bodyData = Buffer.from(JSON.stringify(req.body));
   }
 
-  if (bodyData) {
-    proxyReq.setHeader('Content-Length', Buffer.byteLength(bodyData));
```

Comment:
this is one piece of behavior that did go away - I didn't see any tests around content length specifically, so I thought this might have just been a best practice with the rewrite that was used here

if this creates a gap, let me know and I can see about doing something similar with the proxy request options

Reply:
Thanks. I believe this was required given the way the previous body was parsed and then re-written into the subsequent request. Given that you're now using the `buffer` option on the `HttpProxy` itself, perhaps it's no longer needed. 

The main behaviour we'll need to check is if we can intercept and modify the incoming body using the hooks/filters. I'll pull this in locally and do some testing, I don't think it's an area with great test coverage if I'm honest.